pkg/providers/tinkerbell/stack/stack.go

```diff
@@ -0,0 +1,212 @@
+package stack
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"sigs.k8s.io/yaml"
+
+	"github.com/aws/eks-anywhere/pkg/filewriter"
+	"github.com/aws/eks-anywhere/pkg/logger"
+	releasev1alpha1 "github.com/aws/eks-anywhere/release/api/v1alpha1"
+)
+
+const (
+	args              = "args"
+	createNamespace   = "createNamespace"
+	deploy            = "deploy"
+	env               = "env"
+	image             = "image"
+	namespace         = "namespace"
+	overridesFileName = "tinkerbell-chart-overrides.yaml"
+
+	boots          = "boots"
+	hegel          = "hegel"
+	tinkController = "tinkController"
+	tinkServer     = "tinkServer"
+	grpcPort       = "42113"
+
+	// TODO: remove this once the chart is added to bundle
+	helmChartOci     = "oci://public.ecr.aws/h6q6q4n4/tinkerbell"
+	helmChartName    = "tinkerbell"
+	helmChartVersion = "0.1.0"
+)
+
+type Docker interface {
+	ForceRemove(ctx context.Context, name string) error
+	Run(ctx context.Context, image string, name string, cmd []string, flags ...string) error
+}
+
+type Helm interface {
+	InstallChartWithValuesFile(ctx context.Context, chart, ociURI, version, kubeconfigFilePath, valuesFilePath string) error
+}
+
+type Installer struct {
+	docker          Docker
+	filewriter      filewriter.FileWriter
+	helm            Helm
+	namespace       string
+	createNamespace bool
+	bootsOnDocker   bool
+}
+
+type InstallerOption func(s *Installer)
+
+type StackInstaller interface {
+	// WithNamespace(ns string, create bool) *Installer
+	// WithBootsOnDocker() *Installer
+	// WithBootsOnKubernetes() *Installer
+	Install(ctx context.Context, bundle releasev1alpha1.TinkerbellStackBundle, tinkServerIP, kubeconfig string, opts ...InstallerOption) error
+	UninstallLocal(ctx context.Context) error
+}
+
+func WithNamespace(ns string, create bool) InstallerOption {
```

Comment:
this is needed because the cli performs different actions on bootstrap vs workload cluster currently. On bootstrap cluster, it creates the namespace initially but on workload it doesn't create the namespace until much later.
Helm errors out if we try to create a namespace that already exists. So for bootstrap we need to skip ns creation since the cli already creates it. For workload, we do need to create the namespace.

Reply:
Can we make it a separate option given there's only ever 1 namespace at play. This allows consumers to specify the desired behavior independently of specifying the namespace and rely on default behavior around namespace creation without supplying any options.